sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -175,7 +175,11 @@ case class EnsureRequirements(
           child
         case ((child, dist), idx) =>
           if (bestSpecOpt.isDefined && bestSpecOpt.get.isCompatibleWith(specs(idx))) {
-            child
+            bestSpecOpt match {
```

Comment:
This is pushing down the join key to the BatchScanExec of the partitioned side (to do de-duping).

Reply:
Can we add some comments here? to explain this is the case when `isKeyGroupCompatible` is false, but we still want to avoid shuffling on one side?

Reply:
Added comment